### ISSUE TITLE
The "out" parameters are not compiled properly.

### ISSUE THREAD

**Issue by LaoR-UA** [cake-build/cake#1930]
### What You Are Seeing?
When I create alias that accepts out parameters - cake will generate not working code.
For example, the method:

```
public interface IVersion
{
 public int Build {get;}
}

public class CusomVersion : IVersion
{
     public CustomVersion(int build)
     {
            Build = build;
     }
     public int Build {get;}
}

[CakeMethodAlias]
public static bool TryParse(this ICakeContext context, string versionToParse, out IVersion version)
{
.....
}
```

Will generate wrapper:
```
public System.Boolean TryParse(System.String versionToParse, out MyCake.Common.IVersion& version)
{
    return MyCake.Common.VersionHelper.TryParse(Context, versionToParse, out version);
}
```
And this leads to a lot of errors due to "&" symbol at the end of out parameter type declaration.

### What is Expected?
I expect to see properly generated wrapper and able to use "out" parameters.

### What version of Cake are you using?
Cake version is 0.23.0

### Are you running on a 32 or 64 bit system?
I am running on 64 bit Windows 10 system (client version, local machine).

**Comment by jnm2** [comment]
Now that Cake is on Roslyn, what if you generated syntax nodes in code rather than generating text?

**Comment by LaoR-UA** [comment]
I do not generate anything, it was done by Roslin on compile stage.

**Comment by jnm2** [comment]
@LaoR-UA Sorry I wasn't clear; I was addressing the Cake team.

**Comment by devlead** [comment]
Fixed by #1934
